src/display/bitmap.cpp

```diff
@@ -1086,12 +1086,18 @@ void Bitmap::stretchBlt(IntRect destRect,
     SDL_Surface *blitTemp = 0;
     bool touchesTaintedArea = p->touchesTaintedArea(destRect);
     bool unpack_subimage = srcSurf && gl.unpack_subimage;
-    
+
+    const bool scaleIsOne = sourceRect.w == destRect.w && sourceRect.h == destRect.h;
+    if (scaleIsOne) {
+        smooth = false;
+    }
+
     if (!srcSurf && opacity == 255 && !touchesTaintedArea)
     {
         /* Fast blit */
-        GLMeta::blitBegin(getGLTypes());
-        GLMeta::blitSource(source.getGLTypes());
+        // TODO: Support smooth scaling other than just Bilinear.
+        GLMeta::blitBegin(getGLTypes(), false, SameScale);
+        GLMeta::blitSource(source.getGLTypes(), SameScale);
```

Comment:
Smooth scaling for `stretch_blt` wasn't supposed to work, but it was accidentally working when smooth scaling for the framebuffer was enabled. And this part of the PR breaks that. So I guess I need to implement an alternative, otherwise the perceived graphical quality will decrease.